src/Dropdown.tsx

```diff
@@ -156,8 +165,20 @@ const Dropdown: BsPrefixRefForwardingComponent<'div', DropdownProps> =
         if (
           event.currentTarget === document &&
           (source !== 'keydown' || event.key === 'Escape')
-        )
+        ) {
           source = 'rootClose';
+
+          const noOuterCloseModes =
```

Comment:
Might be worth leaving comments about click what elements triggers the closing of the dropdown so it's more clear